Sources/TensorFlow/Operators/LinearAlgebra.swift

```diff
@@ -177,6 +177,25 @@ public func trace<T: TensorFlowNumeric>(_ matrix: Tensor<T>) -> Tensor<T> {
     return matrix.diagonalPart().sum(squeezingAxes: -1)
 }
 
+/// Computes the determinant of an optionally batched matrix.
+/// 
+/// - Parameter matrix: A tensor of shape `[..., M, M]`.
+/// - Returns: A tensor containing the determinants of all input submatrices.
+@inlinable 
+func det<T: TensorFlowFloatingPoint>(_ matrix: Tensor<T>) -> Tensor<T> {
+    _Raw.matrixDeterminant(matrix)
+}
+
+/// Computes the sign and the natural logarithm of the absolute value of the determinant of an
+/// optionally batched square matrix.
+///
+/// - Parameter matrix: A tensor of shape `[..., N, M, M]`.
+/// - Returns: A tensor containing the determinants of all input submatrices.
+@inlinable
+func slogdet<T: TensorFlowFloatingPoint>(_ matrix: Tensor<T>) -> (Tensor<T>, Tensor<T>) {
```

Comment:
When evaluating API names, I pretend that I'm encountering the API for the first time:
* Is the function name easily understandable?
* If the function returns a tuple of values, is the meaning of each value clear?
  * I think tuple element labels are often useful for providing additional meaning.

---

I think the name `slogdet` has room for improvement:
* The name is domain-specific and doesn't make sense to people unfamiliar with `numpy.linalg.slogdet` or `tf.linalg.slogdet`.
* The returned tuple has no element labels, so the meaning of each tuple element is not clear.

---

How about (make sure to wrap to 100 columns):
```suggestion
func signAndLogDeterminant<T: TensorFlowFloatingPoint>(_ matrix: Tensor<T>) -> (sign: Tensor<T>, logDeterminant: Tensor<T>) {
```

Reply:
Actually, how about just the name `func logDeterminant` instead of `func signAndLogDeterminant`? The result element labels sufficiently distinguish the sign and log det results.

Reply:
Obsolete. Let's continue to use `slogdet` after discussion in https://github.com/tensorflow/swift-apis/pull/604#issuecomment-573653157.